controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorGroupImpl.java

```diff
@@ -204,8 +199,41 @@ protected void shutDown() {
         synchronized (lock) {
             long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "shutDown");
             try {
-                cleanUpCheckpointStore();
+                // If any of the following operations error out, the fact is just logged.
+                // And some other controller process would clean-up the reader group data
+                // its readers and their position objects from checkpoint store.
+
+                // Seal the Reader Group
+                Map<String, Position> readerPositions = ImmutableMap.of();
+                try {
+                    readerPositions = sealReaderGroup(actorSystem.getProcess());
+                } catch (CheckpointStoreException e) {
+                    log.warn("Error sealing Reader Group {}.", readerGroup.getGroupName(), e);
```

Comment:
Code coverage needs to be improved for this PR. In the codecov report, most of these catch blocks are never exercised. As the issue that this PR is fixing happens in the non-happy path, I suspect that is when these catch blocks will get exercised. My suggestion: i) break down the different steps in `shutDown()` into individual functions, ii) make them package protected and tag them with `@VisibleForTesting` so it is easier to test them individually, iii) for testing the whole workflow of `shutDown()`, you can `spy()` an `EventProcessorGroupImpl` object an make specific methods, like `sealReaderGroup`, throw whatever exception and see how the rest of the shutdown workflow behaves in the presence of that event.

Reply:
working on coverage...

Reply:
fixed